ipns/ipns.go

```diff
@@ -94,15 +97,14 @@ func (m *Manager) Publish(pth path.Path, keyID string) {
 	for {
 		select {
 		case ptl <- struct{}{}:
-			pctx, cancel := context.WithCancel(context.Background())
+			pctx, cancel := context.WithTimeout(context.Background(), publishTimeout)
 			m.ctxsLock.Lock()
 			m.ctxs[keyID] = cancel
 			m.ctxsLock.Unlock()
 			if err := m.publishUnsafe(pctx, pth, keyID); err != nil {
 				if !errors.Is(err, context.Canceled) {
 					// Logging as a warning because this often fails with "context deadline exceeded",
 					// even if the entry can be found on the network (not fully saturated).
-					// The publish deadline seems to be fixed at one minute. ¯\_(ツ)_/¯
```

Comment:
They fixed it: https://github.com/ipfs/go-ipfs/pull/7256. Now we can use a longer timeout since these are just spinning on their own anyway.